### PR TITLE
E1861. Improving search facility in Expertiza

### PR DESCRIPTION
**Pull request by shants**
Please put the title of your pull request in the following format: *Project ID. Name*
for example, `[Project id/Independent study/Issue fix/...]. project name`, eg., `E904. Integrating courses and assignments on Expertiza with Moodle`

We suggest you include below information in your pull request:
- A description of the changes proposed in the pull request.
- @mentions of the person or team responsible for reviewing proposed changes.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -59,7 +59,21 @@ def set_anonymized_view
...
   def list
     user = session[:user]
-    @users = user.get_user_list
+
+    search_name = ".*"
+    search_fname = ".*"
+    search_id = ".*"
+    search_email = ".*"
+
+    search_name = ".*" + params[:search_name].strip + ".*" if params[:search_name].present?
```
Comment (vismay-golwala) [inline app/controllers/users_controller.rb]: Can we combine the logic into:
```
if params[:search_name].present? (search_name = ".*" + params[:search_name].strip + ".*") : (search_name = ".*")
```

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -59,7 +59,21 @@ def set_anonymized_view
...
+
```
Comment (vismay-golwala) [inline app/controllers/users_controller.rb]: Not needed blank space

File: app/models/assignment_node.rb [hunk 3]
```diff
@@ -38,7 +38,66 @@ def self.get(sortvar = nil, sortorder = nil, user_id = nil, show = nil, parent_i
...
+    created_until = search[:created_until].to_s.strip
+
+    associations = {assignment: [:due_dates]}
+
+    associations[:assignment] << {participants: :user} if participant_name.present? || participant_fullname.present?
+
+    query = self.includes(associations).where(find_conditions)
+
+    query = query.where('assignments.name LIKE ?', "%#{name}%") if name.present?
+
```
Comment (vismay-golwala) [inline app/models/assignment_node.rb]: Remove whitespaces above and add comments below.

**Comment by dharmangbhavsar** [comment]
After advanced search, either:

1. Keep the box open
2. Clear the queries in advanced search box

**Pull request closed without merging**